### PR TITLE
setCompliment cor algorithms

### PR DESCRIPTION
**Pull request by aaronev**

### CODE REVIEW

File: src/collatzConjecture.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+			array.push(n)
+		}
+		else {
+			n = (3 * array[i]) + 1 
+			array.push(n)
+		}
+		i++
+	}
+	return array
+}
```
Comment (DianaVashti) [inline src/collatzConjecture.js]: I would not indent so much on nested lines but the code and logic looks good 👍

File: src/factorial.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default function factorial(num) {
+	if (num < 0) return 'error'
+	var result = num
+	for (var j = 1; j < num; j++) {
+		 result *= j
+	}
+	return result
+}
```
Comment (DianaVashti) [inline src/factorial.js]: Cool implementation!

File: src/fibonacci.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		}
+	if (num > 1) {
+		array.push(1);
+	}
+	for (var i = 2; i < num; i++) {
+		var result = array[i-2] + array[i-1];
+		array.push(result);
+	}
+	return array
+}
```
Comment (DianaVashti) [inline src/fibonacci.js]: 👍

File: src/fizzBuzz.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+			continue
+		}
+		if (i % 3 == 0) {
+			array.push('Fizz')
+			continue
+		}
+		array.push(i)
+	}
+	return array
+}
```
Comment (DianaVashti) [inline src/fizzBuzz.js]: 👍

File: src/isPalindrome.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export default function isPalindrome(string) {
+	var master = string.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"");
+	var compare = master.split("").reverse().join("");
+	if(master == compare) {
+		return true
+	}
+	return false;
+}
```
Comment (DianaVashti) [inline src/isPalindrome.js]: 👍

File: src/isPalindrome.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export default function isPalindrome(string) {
+	var master = string.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"");
+	var compare = master.split("").reverse().join("");
+	if(master == compare) {
```
Comment (DianaVashti) [inline src/isPalindrome.js]: why not `===`?

File: src/makeChange.js [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+  while (change - 5 >= 0) {
+  	coins.nickels++;
+  	change -= 5;
+  }
+  while (change - 1 >= 0) {
+  	coins.pennies++;
+  	change -=1;
+  }
+  return coins;
 }
```
Comment (DianaVashti) [inline src/makeChange.js]: 👍  but consider this refactor:
```
export default function makeChange({price, amountGiven}) {
  var changeStart = amountGiven - price
  var quarterCount = Math.floor(changeStart/25)
  var dimeCount = Math.floor((changeStart%25)/10)
  var nickelCount = Math.floor(((changeStart%25)%10)/5)
  var pennyCount = Math.floor(((changeStart%25)%10)%5)

    return {
      quarters: quarterCount,
      dimes: dimeCount,
      nickels: nickelCount,
      pennies: pennyCount
    }
}
```

File: src/setCompliment.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+export default function setCompliment(a, b) {	
+	var array = [],
+	if (typeof a !== 'object' || typeof b !== 'object') {
+		return 'error'
+	}
+	else if (a.length !== b.length) {
+		return 'not the same length'
```
Comment (DianaVashti) [inline src/setCompliment.js]: I don't think it's necessary the arrays be the same length?

File: src/setCompliment.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+		if (b.find(matches) !== undefined) {
+			array.push(b.find(matches))
+		}
+	}
+	for (var i = 0; i < array.length; i++) {
+		var pos = b.indexOf(array[i])
+		b.splice(pos, 1)
+	}
+	return b
+}
```
Comment (DianaVashti) [inline src/setCompliment.js]: I would re-do this one so that the arrays can be different lengths.

File: test/collatzConjecture_test.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import { expect } from "chai"
+import collatzConjecture from "../src/collatzConjecture"
+
+describe("collatzConjecture", function() {
+
+	it("should only return an array with 1 inside", function() {
```
Comment (DianaVashti) [inline test/collatzConjecture_test.js]: 1 what? index, item, number?

Comment (DianaVashti) [inline test/collatzConjecture_test.js]: consider ```it("if given an input 1, returns an array with only the number 1", function() {```

File: test/collatzConjecture_test.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+		const output = collatzConjecture(1)
+		expect(output).to.deep.equal([1])
+	})
+
+	it("should return an array of numbers", function() {
```
Comment (DianaVashti) [inline test/collatzConjecture_test.js]: "should return an array of numbers with the last index equaling 1"

File: test/factorial_test.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import { expect } from "chai"
+import factorial from "../src/factorial"
+
+ describe("factorial", function() {
+
+ 	it("should return 120 from 5", function() {
```
Comment (DianaVashti) [inline test/factorial_test.js]: "Given the input 5, the function returns the factorial of 5"

File: test/factorial_test.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+ 	it ("should return an error when trying a negative number", function() {
+ 		const output = factorial(-1)
+ 		expect(output).to.equal("error")
+ 	})
+
+ 	it("should return 5040 from 7", function() {
+ 		const output = factorial(7)
+ 		expect(output).to.equal(5040)
+ 	})
```
Comment (DianaVashti) [inline test/factorial_test.js]: This test is the exact same as far as functionality as the first test. What about testing for a negative number?

File: test/isPalindrome_test.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+
+	it('should be true if string is the same in reverse', function() {
+		const output = isPalindrome('radar')
+		expect(output).to.equal(true)
+	})
+	it('should be false if string is not the same in reverse', function() {
+		const output = isPalindrome('hello')
+		expect(output).to.equal(false)
+	})
+	it('should be true if string has spacing and punctuation', function() {
```
Comment (DianaVashti) [inline test/isPalindrome_test.js]: "Spacing and punctuation should be ignored when testing truthyness"

File: test/setCompliment_test.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+		const a = [1, 2, 3, 4]
+		const b = [2, 4, 6, 8]
+		expect(setCompliment(a, b)).to.be.deep.equal([6,8])
+	})
+	
+	it('should return an error if lengths are not even', function() {
+		const a = [1, 2, 3, 4]
+		const b = [2, 4, 6, 8, 9, 10]
+		expect(setCompliment(a, b)).to.be.deep.equal('not the same length')
+	})
```
Comment (DianaVashti) [inline test/setCompliment_test.js]: See above comment re: arrays being the same length

File: test/setIntersection_test.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+		expect(setIntersection(a, b)).to.equal("There are no numbers in common")
+	})
+
+	it ("should return an error", function(){
+		const a = []
+		const b = "str"
+		expect(setIntersection(a, b)).to.equal("error")
+	})
+
+})
```
Comment (DianaVashti) [inline test/setIntersection_test.js]: The second two tests could have clearer it statements

File: test/setUnion_test.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+		const a = [1, 2, 3, 4]
+		const b = [2, 4, 6, 8]
+		expect(setUnion(a, b)).to.deep.equal([1, 2, 3, 4, 6, 8]) 
+	})
+
+	it('should return b if a is null', function() {
+		const b = null
+		const a = [2, 4, 6, 8]
+		expect(setUnion(a, b)).to.deep.equal([2, 4, 6, 8]) 
+	})
```
Comment (DianaVashti) [inline test/setUnion_test.js]: What about if a is null?

Comment (DianaVashti) [inline test/setUnion_test.js]: If this tests for if either are null I would make the it statement more clear. If it does not I would add another test for if a is null.
